backend/x86_binary_emitter.ml

```diff
@@ -1177,8 +1258,58 @@ let emit_pop b dst =
   | (Mem _ | Mem64_RIP _) as rm -> emit_mod_rm_reg b no_rex [ 0x8F ] rm 0
   | _ -> assert false
 
+let emit_popcnt b ~dst ~src =
+  match (dst, src) with
+  | (Reg16 reg, ((Reg16 _ | Mem _ | Mem64_RIP _) as rm))
+  | (Reg32 reg, ((Reg32 _ | Mem _ | Mem64_RIP _) as rm)) ->
+    (* POPCNT r16, r/m16 and POPCNT r32, r/m32 *)
+    buf_int8 b 0xF3;
+    emit_mod_rm_reg b no_rex [ 0x0F; 0xB8 ] rm (rd_of_reg64 reg);
+  | (Reg64 reg, ((Reg64 _ | Mem _ | Mem64_RIP _) as rm)) ->
+    (* POPCNT r64, r/m64 *)
+    buf_int8 b 0xF3;
+    emit_mod_rm_reg b rexw [ 0x0F; 0xB8 ] rm (rd_of_reg64 reg);
+  | _ -> assert false
+
+let rd_of_prefetch_hint = function
+  | Nta -> 0
+  | T0 -> 1
+  | T1 -> 2
+  | T2 -> 3
+
+let emit_prefetch b ~is_write ~hint rm =
+  match (is_write, hint, rm) with
+  | (false, _, (Mem _ | Mem64_RIP _)) ->
+    (* PREFETCHT0 m8, PREFETCHT1 m8, PREFETCHT1 m8 and PREFETCHNTA m8 *)
+    emit_mod_rm_reg b no_rex [ 0x0F; 0x18 ] rm (rd_of_prefetch_hint hint)
+  | (true, (T0 | T1), (Mem _ | Mem64_RIP _)) ->
+    (* PREFETCHW m8 and PREFETCHWT1 m8 *)
+    (* X86_gas seems to also emit prefetchwt1 if hint is [T2 | Nta], not sure
+       whether such instruction is ever produced. If so I assume we should
+       behave similarly. *)
+    emit_mod_rm_reg b no_rex [ 0x0F; 0x0D ] rm (rd_of_prefetch_hint hint)
```

Comment:
Why not match what `x86_gas` does and emit "0F 0D /2" for `T1 | T2 | Nta`?

Reply:
If that's the intent behind such values then sure! I was a bit surprised all these could come up considering it maps to the same instruction in the end but it's probably better to be consistent with `X86_gas` here!